### PR TITLE
feat: [FFM-3559]: use custom NoData component for environments empty …

### PR DESCRIPTION
**Pull request by crystal-fong**
Updating the empty state of Environments, Targets and Target Groups page. 
Each uses the custom `NoData` component, with updated image, heading, description and link to the relevant Harness Docs

#### Screenshots

Old environment empty state:
image

New environment empty state: 
image

link: https://docs.harness.io/article/1j7pdkqh7j-create-a-feature-flag#step_2_create_an_environment

Old targets empty state:
image

New targets empty state:
image

link: https://docs.harness.io/article/dbk9uoaid3-add-targets#step_add_a_target

Old target group empty state:
image

New target group empty state:
image

link: https://docs.harness.io/article/5qz1qrugyk-add-target-groups#manage-target-groups

NA

#### PR Checklist
When there are no environments created, Targets, Target Groups, Environments Page would show the updated empty-state

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/pages/target-management/targets/NoTargetsView.tsx [hunk 1]
```diff
@@ -7,24 +7,27 @@
...
+import { String, useStrings } from 'framework/strings'
 import { NewTargets } from './NewTarget'
 import imageURL from './target.svg'
 
 export interface NoTargetsViewProps {
   onNewTargetsCreated: () => void
-  hasEnvironment: boolean
 }
 
-export const NoTargetsView: React.FC<NoTargetsViewProps> = ({ onNewTargetsCreated, hasEnvironment }) => {
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/target-management/targets/NoTargetsView.tsx:21]: removed the `hasEnvironment` prop as there's already logic in TargetsPage to determine whether to show an empty environment view or an empty targets view

**Comment by crystal-fong** [comment]
retrigger jest

**Comment by crystal-fong** [comment]
retrigger jest

**Review by laura-mannus — DISMISSED** [review]
Lovely tests as well!

File: src/modules/75-cf/components/NoEnvironment/NoEnvironment.tsx [hunk 2]
```diff
@@ -6,46 +6,26 @@
...
-import { useStrings } from 'framework/strings'
+import type { ContainerProps } from '@harness/uicore/dist/components/Container/Container'
+import { String, useStrings } from 'framework/strings'
 import EmptyContent from './EmptyContent.svg'
 import EnvironmentDialog, { EnvironmentDialogProps } from '../CreateEnvironmentDialog/EnvironmentDialog'
+import { NoData } from '../NoData/NoData'
 
 export interface NoEnvironmentProps extends ContainerProps {
   onCreated: EnvironmentDialogProps['onCreate']
 }
```
Comment (knagurski) [inline src/modules/75-cf/components/NoEnvironment/NoEnvironment.tsx:17]: Looks like you're not using any of the members from `ContainerProps`, so you can simplify.
```suggestion
export interface NoEnvironmentProps {
  onCreated: EnvironmentDialogProps['onCreate']
}
```

File: src/modules/75-cf/components/NoEnvironment/EmptyContent.svg [hunk 3]
```diff
@@ -1,3 +1 @@
...
....0515-.206-.1451-.226-.1721-.046-.0626-.067-.1226-.071-.1351v-.0012c-.008-.0221-.013-.0412-.016-.0529-.011-.0426-.018-.0901-.023-.1283-.004-.0344-.008-.0754-.013-.1226-.851-2.3104-2.141-3.9768-3.71-5.5119-.198-.1931-.201-.5097-.008-.7071s.51-.2008.707-.0077c1.19 1.1637 2.249 2.4233 3.09 3.9768.092-.7303.254-1.5926.535-2.5442.999-3.3935 3.486-7.898 9.473-11.589.235-.1449.543-.0719.688.1632z" fill="#000" fill-rule="evenodd"/><path clip-rule="evenodd" d="m274.72 36.1694c.989.0929 1.944.3073 2.791.6349 1.444.5591 3.253 2.0137 4.394 4.0302 1.152 2.0382 1.636 4.6786.351 7.5613-1.158 2.5988-3.085 4.5209-5.414 5.4426-2.335.9241-5.034.826-7.688-.5458-2.694-1.3922-4.391-3.8045-4.919-6.825-.347-1.9906.094-4.5649 1.717-6.6479.837-1.0733 1.976-2.0033 3.46-2.6565.035-.1568.144-.2941.305-.3575 1.505-.5946 3.289-.7938 4.978-.6387.009.0006.017.0014.025.0024zm-.083.9965c-3.993.0057-6.477 1.4471-7.896 3.2685-1.437 1.8437-1.823 4.1274-1.521 5.8613.477 2.73 1.992 4.8677 4.393 6.1085 2.413 1.2472 4.811 1.3158 6.861.5044 2.057-.8138 3.804-2.5292 4.869-4.9198 1.135-2.5486.713-4.8545-.309-6.6621-1.034-1.8293-2.67-3.1198-3.884-3.5898-.759-.2937-1.62-.4864-2.513-.571z" fill="#000" fill-rule="evenodd"/></svg>
```
Comment (knagurski) [inline src/modules/75-cf/components/NoEnvironment/EmptyContent.svg:1]: Can you remove the `width` and `height` and run through an optimiser (I like ImageOptim). The SVG will then scale with it's parent, so you can control it with the `imageWidth` prop of `NoData`

File: src/modules/75-cf/images/segment.svg [hunk 4]
```diff
@@ -1,3 +1 @@
...
...1-.74.172-.237.385-.358.639-.364.1-.002.233.039.401.123.168.085.332.231.492.437.171.206.316.48.434.82.13.34.201.759.213 1.257z" fill="#0092e4"/></g><path d="m172.184 57.9605c-4.93-17.7058-16.45-33.6755-31.834-43.698-16.301-10.61884-37.611-13.809464-56.6754-14.236868-26.6217-.609631-60.3435 9.614938-73.4407 35.000768-1.09 2.1105 2.1205 3.9228 3.2105 1.8123 9.2804-17.9907 31.4457-28.38094 50.5696-31.78693 20.5717-3.65778 43.065-.78191 62.53 6.47733 20.323 7.5773 34.434 24.9717 41.369 45.0961 7.72 22.3907 3.446 44.6888-8.876 64.3558-25.439 40.594-87.8731 53.77-125.8425 23.004-17.3944-14.095-28.7388-32.95-28.95747-55.5032-.12922-13.2926 2.46172-28.1358 9.09807-39.7983 2.0575-3.6114 4.8207-6.706 7.9385-9.499.0463-.7256.0994-1.4446.1491-2.1602-.3612-.4606-.6693-.9576-.8283-1.5738-.0166-.0696-.0133-.1226-.0299-.1955-1.8951 1.5937-3.7306 3.2668-5.4668 5.0659-7.59053 7.8556-10.58236 18.7528-12.69619 29.1596-4.20115 20.6744-1.706305 40.0035 10.14839 57.4705 10.1483 14.956 25.8 28.875 43.5058 33.736 21.2708 5.837 43.4792 5.344 63.6073-3.671 9.108-4.076 18.981-8.078 26.635-14.658 9.386-8.078 16.436-19.485 22.129-30.336 8.677-16.5394 8.67-36.4253 3.757-54.0615z" fill="#000" opacity=".2"/></svg>
```
Comment (knagurski) [inline src/modules/75-cf/images/segment.svg:1]: Same with this one (actually same with all SVGs, avoid the `width` and `height`)

File: src/modules/75-cf/pages/target-management/segments/__tests__/SegmentsPage.test.tsx [hunk 5]
```diff
@@ -234,7 +240,9 @@ describe('SegmentsPage', () => {
       </TestWrapper>
     )
 
-    expect(container).toMatchSnapshot()
+    expect(document.querySelector('[data-testid="nodata-image"]')).toBeInTheDocument()
+    expect(screen.getByRole('button', { name: 'plus cf.segments.create' })).toBeInTheDocument()
+    expect(screen.getByRole('heading', { name: 'cf.segments.noTargetGroupsForEnv' })).toBeInTheDocument()
```
Comment (knagurski) [inline src/modules/75-cf/pages/target-management/segments/__tests__/SegmentsPage.test.tsx:245]: Loving the use of roles!

File: src/modules/75-cf/pages/target-management/targets/target.svg [hunk 6]
```diff
@@ -1,3 +1 @@
...
...rg/2000/svg"><g opacity=".7"><path d="m47.5 135.5c-16.8.8-36.6667-6-44.5-9.5l32.5-78c.6667 0 4 .3 12 1.5s15.3333-.5 18-1.5l26.5 78c-7.8333 2.833-27.7 8.7-44.5 9.5z" fill="#f3f3fa"/><g stroke-linecap="round" stroke-linejoin="round" stroke-width="1.54701"><path d="m42.021 5.02681c-25.6095 13.53659-12.948 32.26389-11.1701 34.45679 6.7314 8.3024 19.0649 9.2938 29.6959 9.7764 9.3818.4259 14.8233-5.9463 17.4022-14.3912 3.0049-9.84 1.3583-21.0569-6.8621-27.80807-7.881-6.472352-15.6942-7.210207-25.2762-4.78479-9.3207 2.35931-18.9979 10.04866-22.0674 19.68956-3.4085 10.7053 2.164 19.6943 11.0574 24.9017 9.2504 5.4164 29.4241 4.9223 37.5821-2.034" stroke="#000"/><path d="m36.8846 48.1794c-10.6154 31.5686-21.2308 54.6376-35.3846 82.5646" stroke="#000"/><path d="m65.1921 43.4617c.1357.587.1933.8997.3825 1.4701 9.8851 29.8002 16.347 51.1453 27.9252 83.4532" stroke="#000"/><path d="m64.5 51c.115.4908.1639.7522.3243 1.2291 8.3808 24.9145 13.8594 42.76 23.6757 69.7709" stroke="#6b6d85"/><path d="m1.5 124.846c17.6923 12.975 69.5897 16.513 92 1.18" stroke="#000"/><path d="m34.2912 29.6738c1.803 5.0529 5.5563 8.3924 11.0632 9.2231 4.3538.6567 8.5882-.9523 11.1796-4.3563" stroke="#000"/></g></g></svg>
```
Comment (knagurski) [inline src/modules/75-cf/pages/target-management/targets/target.svg:1]: Same here

**Review by knagurski — COMMENTED** [review]
Some really great work. Just a few things to look at

File: src/modules/75-cf/strings/strings.en.yaml [hunk 7]
```diff
@@ -301,6 +299,8 @@ targets:
   deleteTargetSuccess: Target <strong>{{name}}</strong> deleted.
   pageDescription: This list displays targets manually added or impacted by flags.
   uploadError: CSV processing failed
+  noTargetForEnv: There are no targets in this environment
```
[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/strings/strings.en.yaml:302]: good spot!

**Review by laura-mannus — DISMISSED** [review]

File: src/modules/75-cf/strings/strings.en.yaml [hunk 8]
```diff
@@ -301,6 +299,8 @@ targets:
   deleteTargetSuccess: Target <strong>{{name}}</strong> deleted.
   pageDescription: This list displays targets manually added or impacted by flags.
   uploadError: CSV processing failed
+  noTargetForEnv: There are no Targets in this environment
+  noTargetDescription: <a href="https://docs.harness.io/article/dbk9uoaid3-add-targets#step_add_a_target" target="_blank" rel="noreferrer">Targets</a> are individual evaluations of flags based on attributes you define in our SDKs. For instance, a target can be a single customer or account that toggle a flag.
```
Comment (knagurski) [inline src/modules/75-cf/strings/strings.en.yaml:303]:
```suggestion
  noTargetForEnv: There are no Targets in this Environment
  noTargetDescription: <a href="https://docs.harness.io/article/dbk9uoaid3-add-targets#step_add_a_target" target="_blank" rel="noreferrer">Targets</a> are individual evaluations of Flags based on attributes you define in our SDKs. For instance, a Target can be a single customer or account that toggle a Flag.
```

File: src/modules/75-cf/strings/strings.en.yaml [hunk 9]
```diff
@@ -359,7 +359,8 @@ segments:
   nameRequired: Target Group name is required
   idRequired: Target Group identifier is required
   noSegment: There are currently no target groups
-  noSegmentForEnv: There are currently no target groups for this environment
+  noTargetGroupsForEnv: There are no target groups in this environment
+  noTargetGroupsDescription: <a href="https://docs.harness.io/article/5qz1qrugyk-add-target-groups#manage-target-groups" target="_blank" rel="noreferrer">Target Groups</a> are larger collection of targets used to toggle flags for entire cohorts. For instance, a target group could be an entire country, an early adopter list or an internal team.
```
Comment (knagurski) [inline src/modules/75-cf/strings/strings.en.yaml:363]:
```suggestion
  noTargetGroupsForEnv: There are no Target Groups in this Environment
  noTargetGroupsDescription: <a href="https://docs.harness.io/article/5qz1qrugyk-add-target-groups#manage-target-groups" target="_blank" rel="noreferrer">Target Groups</a> are larger collection of Targets used to toggle Flags for entire cohorts. For instance, a Target Group could be an entire country, an early adopter list or an internal team.
```

**Review by knagurski — APPROVED** [review]
Noice 👍🏻

**Review by laura-mannus — APPROVED** [review]

**Pull request merged**